tox.ini

```diff
@@ -32,27 +36,40 @@ description =
     warnings: treating warnings as errors
     cov: with coverage
     xdist: using parallel processing
+allowlist_externals =
+    git
 pass_env =
     CRDS_*
     HOME
     CI
     TOXENV
     CODECOV_*
+set_env =
+    devdeps: UV_INDEX = https://pypi.anaconda.org/liberfa/simple https://pypi.anaconda.org/astropy/simple https://pypi.anaconda.org/scientific-python-nightly-wheels/simple
+    devdeps: UV_INDEX_STRATEGY = unsafe-any-match
 extras =
     test
+uv_resolution =
+    oldestdeps: lowest-direct
 deps =
+    withromancal: romancal @ git+{[main]rcal_repo}
     xdist: pytest-xdist
-    cov: pytest-cov
-    oldestdeps: minimum_dependencies
+    cov: pytest-cov >= 4.1.0
     rad: rad @ git+https://github.com/spacetelescope/rad.git@main#egg=rad
+    romancal: romancal[test] @ git+{[main]rcal_repo}
+    devdeps: -r requirements-dev.txt
+    oldestdeps: minimum_dependencies
+change_dir =
+    romancal,oldestdeps: {env_tmp_dir}
 commands_pre =
     oldestdeps: minimum_dependencies roman_datamodels --filename requirements-min.txt
-    oldestdeps: pip install -r requirements-min.txt
-    devdeps: pip install -r requirements-dev.txt -U --upgrade-strategy eager
-    pip freeze
+    oldestdeps: uv pip install -r requirements-min.txt
```

Comment:
@braingram This forces an exact pin to all the lower pins. Does this do what you wanted?

Reply:
Thanks. Testing locally the `oldestdeps` now fails if one of the lower pins is unresolvable.